CMakeLists.txt

```diff
@@ -174,6 +181,72 @@ else()
   set(FIRESTORE_SOURCE_DIR ${FIREBASE_BINARY_DIR}/external/src/firestore)
 endif()
 
+if(DESKTOP)
+  # Use the static versions of the OpenSSL libraries.
+  set(OPENSSL_USE_STATIC_LIBS TRUE)
+  if (MSVC)
+    # Get the correct version of the OpenSSL libraries based on building for MT.
+    if ("${CMAKE_CXX_FLAGS_DEBUG}" MATCHES "/MT" OR
+        "${CMAKE_CXX_FLAGS_RELEASE}" MATCHES "/MT")
+      set(OPENSSL_MSVC_STATIC_RT TRUE)
+    else()
+      set(OPENSSL_MSVC_STATIC_RT FALSE)
+    endif()
+  endif()
+
+  # Use BoringSSL instead of OpenSSL.
+  set(BORINGSSL_ROOT_DIR ${CMAKE_BINARY_DIR}/external/src/boringssl/src CACHE STRING "" FORCE)
+  set(BORINGSSL_BINARY_DIR ${CMAKE_BINARY_DIR}/external/src/boringssl-build CACHE STRING "" FORCE)
+  set(OPENSSL_ROOT_DIR ${BORINGSSL_ROOT_DIR} CACHE STRING "" FORCE)
+
+  # The call below to build_external_dependencies will make sure that these
+  # libraries exist before the libraries are imported via add_library.
+  if (MSVC)
+    set(OPENSSL_SSL_LIBRARY ${BORINGSSL_BINARY_DIR}/ssl/${CMAKE_BUILD_TYPE}/ssl.lib CACHE FILEPATH "" FORCE)
```

Comment:
Have we tested what happens if CMAKE_BUILD_TYPE is not specified?
Like, if we run 
`cmake ..` and not `cmake .. -DCMAKE_BUILD_TYPE=(Release|Debug)`

Reply:
If cmake doesn't set this variable, maybe we should set it to default value ourselves to ensure it always has a value.

Reply:
On Windows it seems to always be set. I'll set it to a default value instead - my research indicates that if it's not set on Windows, it's treated as debug.

Reply:
Hmm... I think I did a quick test on linux and noticed it wasn't set. I thought cmake lets the generator use whatever their default is. But we are safe for now as we provide this value explicitly in our build steps but something to watch out for. Marking it resolved for now.